eclass/cargo.eclass

```diff
@@ -162,12 +166,18 @@ ECARGO_VENDOR="${ECARGO_HOME}/gentoo"
 # group, and then switch over to building with FEATURES=userpriv.
 # Or vice-versa.
 
-# @FUNCTION: cargo_crate_uris
+# @ECLASS_VARIABLE: CARGO_CRATE_URIS
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# List of URIs to put in SRC_URI created from CRATES variable.
```

Comment:
GIT_CRATES is currently not `@PRE_INHERIT` and mergely expected to be set "before cargo_crate_uris is called", meaning variable may be unexpectedly underpopulated.

e.g. rustup sets it just before calling cargo_crate_uris

(does feel very inconsistent with CRATES though)

Reply:
There are also some ebuilds misusing the CRATES variable, especially in EAPI=7 given it used to *not* be PRE_INHERIT, and it only got enforced in EAPI=8

Reply:
> There are also some ebuilds misusing the CRATES variable, especially in EAPI=7 given it used to _not_ be PRE_INHERIT, and it only got enforced in EAPI=8

Then again this one doesn't matter much given if switch to the variable, they should also fix themselves.

Reply:
Given it's just one ebuild and I don't think we really have a good case for doing it differnetly, I'll just mark it pre-inherit for consistency.